packages/cli-hydrogen/src/cli/services/build.ts

```diff
@@ -32,6 +32,26 @@ async function build({directory, targets, base}: DevOptions) {
             })
             // eslint-disable-next-line @typescript-eslint/no-explicit-any
           } catch (error: any) {
+            const hasTailwindConfig =
+              (await file.read(path.join(directory, 'tailwind.config.js'))) ||
+              (await file.read(path.join(directory, 'tailwind.config.ts')))
+
+            if (
+              // eslint-disable-next-line no-restricted-syntax
+              error.message === "Cannot read properties of undefined (reading 'config')" &&
+              directory !== process.cwd() &&
+              hasTailwindConfig
+            ) {
+              const tailwindError = new kitError.Abort(
+                output.content`Running ${output.token.genericShellCommand(
+                  'shopify hydrogen build',
+                )} using a ${output.token.genericShellCommand(
+                  `--path`,
+                )} flag is not supported for projects using Tailwind.css.`,
+              )
+              throw tailwindError
+            }
```

Comment:
I recommend turning this into models and logic that feels less scripty. Here's an example:

```ts
// Loading
const subject: HydrogenStorefront = try loadHydrogenStorefront()

// Validation
validate(subject)

// Execution
build(subject)
```

The storefront model can have an attribute whether or not a Tailwind configuration file exists and that's the information the validator uses to fail. Think of loading the model as taking a snapshot of what's in the system. Once the snapshot has been loaded, you pass it onto the next action in the pipeline to work with that object that you've loaded.

Reply:
Needless to say, unit testing is easier because validators are pure functions that take an in-memory representation as input.